tests/sentry/web/frontend/test_react_page.py

```diff
@@ -261,31 +261,71 @@ def test_handles_unknown_url_name(self):
         assert response.status_code == 200
         self.assertTemplateUsed(response, "sentry/base-react.html")
 
-    def test_customer_domain_non_member_org_superuser(self):
-        org = self.create_organization(owner=self.user)
+    def test_customer_domain_non_member(self):
+        self.create_organization(owner=self.user)
         other_org = self.create_organization()
 
-        self.login_as(self.user, superuser=True)
+        self.login_as(self.user)
+        with override_settings(SENTRY_USE_CUSTOMER_DOMAINS=True), self.feature(
+            {"organizations:customer-domains": [other_org.slug]}
+        ):
+            # Should not be able to induce activeorg
+            assert "activeorg" not in self.client.session
+            response = self.client.get(
+                "/",
+                HTTP_HOST=f"{other_org.slug}.testserver",
+                follow=True,
+            )
+            assert response.status_code == 200
+            assert response.redirect_chain == [(f"http://{other_org.slug}.testserver/issues/", 302)]
+            assert "activeorg" not in self.client.session
 
-        with self.feature({"organizations:customer-domains": [org.slug]}):
+    def _run_customer_domain_elevated_privileges(self, is_superuser: bool, is_staff: bool):
+        user = self.create_user("foo@example.com", is_superuser=is_superuser, is_staff=is_staff)
+        org = self.create_organization(owner=user)
+        other_org = self.create_organization()
+
+        self.login_as(user, superuser=is_superuser, staff=is_staff)
+        with override_settings(SENTRY_USE_CUSTOMER_DOMAINS=True), self.feature(
+            {"organizations:customer-domains": [other_org.slug]}
+        ):
             # Induce activeorg
+            assert "activeorg" not in self.client.session
             response = self.client.get(
                 "/",
-                HTTP_HOST=f"{org.slug}.testserver",
+                HTTP_HOST=f"{other_org.slug}.testserver",
                 follow=True,
             )
             assert response.status_code == 200
-            assert response.redirect_chain == [(f"http://{org.slug}.testserver/issues/", 302)]
-            assert self.client.session["activeorg"] == org.slug
+            if is_superuser:
+                assert response.redirect_chain == [
+                    (f"http://{other_org.slug}.testserver/issues/", 302)
+                ]
+            else:
+                assert response.redirect_chain == [
+                    (f"http://{other_org.slug}.testserver/auth/login/{other_org.slug}/", 302)
+                ]
+            assert self.client.session["activeorg"] == other_org.slug
```

Comment:
For staff that don't have access to the organization should we be setting `activeorg` to the org they cannot access?

Reply:
@markstory Yes this is fine because we generally use `activeorg` to know which org slug to use when a user accesses `sentry.io` with no org slug given. 

If the user has no permissions, we just redirect them to the login page.

This is the same convention we use in the customer domain middleware here: https://github.com/getsentry/sentry/blob/f0ee34a650fe90733819f0294fccf5b1088b96c0/src/sentry/middleware/customer_domain.py#L107